### PR TITLE
Tests: Convert test_efault and test_io to LibTest

### PR DESCRIPTION
**Pull request by ADKaster**
These bois were hiding some kernel panics under the hood, so fix those as part of converting the tests.

Kernel 🐞s:

- is_user_range() was checking against 0xC0000000, while the userspace PageDirectories only had an addressible range of 0xwhatever to 0xBE000000, which leaves a gap of 32 MiB. Attempting to mmap MAP_FIXED in that region would panic the kernel.

- TmpFS::write had a possible 64-bit --> 32-bit truncation issue when working with file offsets around INT32_MAX, and would panic the kernel if you lseek'd that far and tried to write. Check for overflow, and against numeric limits of size_t.

### CODE REVIEW

File: Kernel/VM/Range.h [hunk 1]
```diff
@@ -58,3 +58,11 @@ class Range {
 };
 
 }
+
+template<>
+struct AK::Formatter<Kernel::Range> : Formatter<FormatString> {
+    void format(FormatBuilder& builder, Kernel::Range value)
+    {
+        return Formatter<FormatString>::format(builder, "[{}, {})", value.base(), value.base().offset(value.size()));
```
Comment (awesomekling) [inline Kernel/VM/Range.h:66]: This doesn't look right, and will come out as `[foo, bar)`

**Review by awesomekling — CHANGES_REQUESTED** [review]

File: Kernel/VM/Range.h (hunk 1 above)
Comment (alimpfard) [reply Kernel/VM/Range.h:66]: That seems alright to me, is it not "End-exclusive range from foo to bar"? (though it'd probably be nice to also know how big the region is)

Comment (ADKaster) [reply Kernel/VM/Range.h:66]: Right, it was intentional to mean it includes the start but not the end value. I can make it a bit more descriptive tho

Comment (awesomekling) [reply Kernel/VM/Range.h:66]: Usually we print ranges as e.g `0xc0001000-0xc0000fff`, so end inclusive.
Is there some reason do it differently here?

Comment (ADKaster) [reply Kernel/VM/Range.h:66]: Well, no not really. I can make it end inclusive. One thing to think about, the memory manager prints end-exclusive ranges :)

```
[Kernel]: MM: Physical page entries: V0xc2001000 - V0xc2802000
[Kernel]: MM: Low memory range @ P0000000000000000 - P0000000000100000
[Kernel]: MM: Kernel range @ P0000000000100000 - P000000000120a000
[Kernel]: MM: Physical Pages range @ P000000000120b000 - P0000000001a11000
[Kernel]: MM: Super physical region: P0000000000122000 - P0000000000222000
[Kernel]: MM: User physical region: P0000000001a11000 - P000000003ffe0000
```

Comment (awesomekling) [reply Kernel/VM/Range.h:66]: Hey it'd be nice to make it print end-inclusive as well :^)

Comment (ADKaster) [reply Kernel/VM/Range.h]: Mmk, my last commit makes those prints look like this instead

```
[Kernel]: MM: Physical page entries: 0xc2001000 - 0xc2801fff (size 0x00801000)
[Kernel]: MM: Low memory range @ P0000000000000000 - P00000000000fffff (size 0x100000)
[Kernel]: MM: Kernel range @ P0000000000100000 - P0000000001209fff (size 0x110a000)
[Kernel]: MM: Physical Pages range @ P000000000120b000 - P0000000001a10fff (size 0x806000)
[Kernel]: MM: Super physical region: P0000000000122000 - P0000000000221fff (size 0x100000)
[Kernel]: MM: User physical region: P0000000001a11000 - P000000003ffdffff (size 0x3e5cf000)
```

**Pull request merged**